src/sas/sascalc/data_util/nxsunit.py

```diff
@@ -191,16 +189,30 @@ def _format_unit_structure(unit=None):
     :param unit: Unit string to be formatted
     :return: Formatted unit string
     """
-    if unit:
-        # Convert a/b^n to a*b^{-n}
-        split = unit.split("/")
-        if len(split) == 1:
-            return unit
-        else:
-            split_ct = split[1].split("^")
-            number = 1 if len(split_ct) == 1 else split_ct[1]
-            split[0] = '' if split[0] == '1' else split[0] + '*'
-            return "{0}{1}^{{-{2}}}".format(split[0], split_ct[0], number)
+    if not unit:
+        return
+    # a-m[ /?]b-n ... -> a^m b^-n and pre_Unit -> preUnit
+    unit = re.sub('([℃ÅA-Za-z ]+)([-0-9]+)', r"\1^\2", unit).replace("_", '')
```

Comment:
This will substitute `a^-b` for `a-b`.  Probably never going to be an issue, but easy to prevent using: `'([℃ÅA-Za-z ]+)(-?[0-9]+)'`?